docs/changelog/82685.yaml

```diff
@@ -0,0 +1,6 @@
+pr: 82685
+summary: Discard intermediate results upon cancellation for stats endpoints
+area: Stats
+type: bug
+issues:
+ - 82337
```

Comment:
@dakrone I this change doesn't qualify for the changelog. It feels like an internal change that doesn't change any user facing functionality apart from having less OOM :p.

Reply:
I think it's worth documenting things like this here. Sometimes folks are resistant to upgrading, but if someone is suffering from the problem that this PR fixes then it's helpful to be able to point to the docs to show that an upgrade would fix the problem.

Reply:
Good point, I will keep it then.